File: Walter/Python_walter/POO/Ejemplo2/circulo.py

import math


class Circulo:

    def __init__(self,r):
        self.__radio = r

    def get_radio(self):
        return self.__radio

    def set_radio(self,r):
        self.__radio = r

    def __str__(self):
        return self.__radio
    
    def perimetro(self):
        return 2 * math.pi * self.__radio
    
    def area(self):
        return math.pi * math.pow(self.__radio,2)
    
    @staticmethod
    def cabecera():
        print('%5s  %10s  %10s  %10s' % ('ID','RADIO','PERIMETRO','AREA'))
        print('%5s  %10s  %10s  %10s' % ('--','-----','---------','----'))

    def cuerpo(self,i):
        print('%5d  %10.2f  %10.2f  %10.2f' % (i,self.get_radio(),self.perimetro(),self.area()))
    
    @staticmethod
    def redondear(numero):
        return round(numero,2)

File: Walter/LENGUAJE_ESTRUCTURADO_12062023-main/APACHENETBEANS/src/poo/circulo/Circulo.java
package poo.circulo;

public class Circulo {

    private double radio;

    public Circulo() {
    }

    public Circulo(double radio) {
        this.radio = radio;
    }

    public double getRadio() {
        return radio;
    }

    public void setRadio(double radio) {
        this.radio = radio;
    }

    public double area() {
        return Math.PI * Math.pow(this.radio, 2);
    }

    public double perimetro() {
        return 2 * Math.PI * this.radio;
    }

    public static void cabecera() {
        System.out.printf("%-3s  %6s  %10s  %11s\n", "N", "RADIO", "AREA", "PERIMETRO");
        System.out.printf("%-3s  %6s  %10s  %11s\n", "-", "-----", "----", "---------");
    }

    public void cuerpo(int i) {
        System.out.printf("%-3d  %6.2f  %10.2f  %11.2f\n", i, this.getRadio(), this.area(), this.perimetro());
    }

}
